### DISCUSSION TITLE
Best way to change the `hideDelay` prop with the default layout?

### DISCUSSION THREAD

**Discussion by miketdonahue** [vidstack/player#1015] [Q&A] [answered]
Hi All,

I'd like to change the `hideDelay` prop on the `Controls` component, but I am unsure of the best way to do so. I do not want to implement my own Controls, I simply want to change the prop to a shorter value. However, I don't see the prop exposed at a higher level, such as the `Player`.

I am using the standard, default layout with the React component.

Is there a way to change the `hideDelay` prop without implementing my own Controls?

Thanks!

**Comment by mihar-22** [comment, marked as answer]
You can change it on the player instance:

```tsx
const player = useRef<MediaPlayerInstance>(null);

useEffect(() => {
  player.current.controls.defaultDelay = 2000;
}, []);

<MediaPlayer ref={player}>
```

Hope that helps! Let me know if there's any issues.

**Reply by mihar-22** [reply]
I've also noted to add this to the docs under the controls component page.

**Reply by miketdonahue** [reply]
Excellent. Thanks so much, @mihar-22!

**Reply by mihar-22** [reply]
Oh just adding here for everyone that there's also the `controlsDelay` prop on the player too (forgot somehow 😓):

```tsx
<MediaPlayer controlsDelay={2000}>
```
